### PR TITLE
Add GET /vectors/list endpoint

### PR DESCRIPTION
**Pull request by xzdshr**
https://docs.pinecone.io/reference/api/data-plane/list

### CODE REVIEW

**Comment by ScotterC** [comment]
Thanks for the PR 🙏! On first glance it looks on the right track. I'll be able to give it more attention later today.

**Comment by ScotterC** [comment]
@xzdshr looks like the lint check was a surprise. Can you run `standardrb --fix` on the code and add as a commit? (I should add that to the readme)

**Comment by xzdshr** [comment]
> @xzdshr looks like the lint check was a surprise. Can you run `standardrb --fix` on the code and add as a commit? (I should add that to the readme)

Done

File: lib/pinecone/vector.rb [hunk 1]
```diff
@@ -37,6 +37,17 @@ def fetch(namespace: "", ids: [])
       self.class.get("#{@base_uri}/vectors/fetch?#{query_string}", options)
     end
 
+    def list(namespace: "", prefix: "", limit: nil, pagination_token: "")
```
[the remark this answers is not in the document]
Comment (xzdshr) [reply lib/pinecone/vector.rb:40]: Yes, I was thinking about the autopagination as well but had some concerns regarding the cases with a huge amount of records in the index, which can lead to a lot of requests and a huge response.
OK, so what's the plan for the current PR?

1. rename to `list_paginated` to match the python client
2. rename to `list_paginated` and implement `list` with autopagination
3. leave it as is for now

[the remark this answers is not in the document]
Comment (ScotterC) [reply lib/pinecone/vector.rb:40]: 3, leave it as is now. It's a barebones client overall and that's fine to leave it that way. 

If tests pass I think we'll be good to go

**Pull request merged**